scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -897,46 +899,17 @@ public List<FilteredTopWorkflow> getTopWorkflowsWithIssues(int numberOfWorkflows
 
         return executeReadOnlyTransaction(session -> {
 
-            StringBuilder queryString = new StringBuilder("select jobName, projectName, (sum(numberOfFailedTasks) + sum(numberOfFaultyTasks)) as errorCount, count(*) as numberOfExecution from JobData where (numberOfFailedTasks > 0 or numberOfFaultyTasks > 0) ");
-
-            boolean hasWorkflowName = !Strings.isNullOrEmpty(workflowName);
-            boolean hasUser = !Strings.isNullOrEmpty(user);
-            boolean hasTenant = !Strings.isNullOrEmpty(tenant);
-
-            if (hasWorkflowName) {
-                queryString.append("and jobName = :workflowName ");
-            }
-            if (hasUser) {
-                queryString.append("and owner = :user ");
-                if (hasTenant) {
-                    queryString.append("and (tenant = :tenant or tenant = null) ");
-                }
-            }
-            if (startTime != 0) {
-                queryString.append("and finishedTime >= :startTime ");
-            }
-            if (endTime != 0) {
-                queryString.append("and finishedTime <= :endTime ");
-            }
-            queryString.append("group by jobName, projectName order by errorCount desc");
-
-            Query query = session.createQuery(queryString.toString());
-            query.setMaxResults(numberOfWorkflows);
-            if (hasWorkflowName) {
-                query.setParameter("workflowName", workflowName);
-            }
-            if (hasUser) {
-                query.setParameter("user", user);
-                if (hasTenant) {
-                    query.setParameter("tenant", tenant);
-                }
-            }
-            if (startTime != 0) {
-                query.setParameter("startTime", startTime);
-            }
-            if (endTime != 0) {
-                query.setParameter("endTime", endTime);
-            }
+            String selectQuery = "select jobName, projectName, (sum(numberOfFailedTasks) + sum(numberOfFaultyTasks)) as errorCount, count(*) as numberOfExecution from JobData where (numberOfFailedTasks > 0 or numberOfFaultyTasks > 0) ";
```

Comment:
could you please add indexes (in JobData class) to numberOfFailedTasks and numberOfFaultyTasks columns ?

Reply:
selectQuery -> selectSubQuery